hal/lib/athena/cpp/priority_mutex.cpp

```diff
@@ -17,12 +19,14 @@ bool priority_recursive_mutex::try_lock() noexcept {
 
 pthread_mutex_t* priority_recursive_mutex::native_handle() { return &m_mutex; }
 
-void priority_mutex::lock() { pthread_mutex_lock(&m_mutex); }
+void hal::priority_mutex::lock() { pthread_mutex_lock(&m_mutex); }
```

Comment:
This and the other function definitions in this file don't need to be namespace qualified.